src/ride_build.cpp

```diff
@@ -182,19 +182,33 @@ bool RideBuildWindow::CanPlaceShop(const ShopType *selected_shop, const XYZPoint
 
 	/* 1. Can the position itself be used to build a shop? */
 	if (_world.GetTileOwner(pos.x, pos.y) != OWN_PARK) return false;
-	auto check_if_location_is_suited = [&pos, selected_shop](const int8 offset, const bool need_flat_ground) {
+	auto check_if_location_is_suited = [&pos, selected_shop](const bool is_flat_ground) {
+		const int8 offset = is_flat_ground ? 0 : -1;
 		const Voxel *vx = _world.GetVoxel(pos + XYZPoint16(0, 0, offset));
-		if (vx == nullptr || vx->GetGroundType() == GTP_INVALID || (need_flat_ground != (vx->GetGroundSlope() == SL_FLAT))) return false;
+		if (vx == nullptr || vx->GetGroundType() == GTP_INVALID) return false;
+		if (is_flat_ground) {
+			if (vx->GetGroundSlope() != SL_FLAT) return false;
+		} else {
+			if (vx->GetGroundSlope() == SL_FLAT) return false;
+			const Voxel *top_voxel = _world.GetVoxel(pos);
+			if (top_voxel != nullptr) {
+				const uint8 slope = top_voxel->GetGroundSlope();
+				if (slope & ISL_BOTTOM_STEEP_NORTH) return false;
+				if (slope & ISL_BOTTOM_STEEP_SOUTH) return false;
+				if (slope & ISL_BOTTOM_STEEP_EAST) return false;
+				if (slope & ISL_BOTTOM_STEEP_WEST) return false;
```

Comment:
This is definitely wrong. `ISL_BOTTOM_STEEP_*` is an imploded slope description (ie enumeration values that define sprite numbers) see tile.h, line 81.
While you can do bit operations on two random integer numbers, it's not likely to do much good.

You can expand the slope to a collection of bits using something like `TileSlope ts = ExpandTileSlope(v->GetGroundSlope());`, `TileSlope` as well as the expand function are in tile.h.

Alternatively, use `IsImplodedSteepSlope(slope)` also in tile.h. A third option is to move the shop one tile higher, on top of the steep slope.

As for code style, here you should compare against not being 0. Only boolean values don't need a comparison.

and last but not least, if you want to test for at least one of several bits, you can merge the comparison with `(v & (B1 | B2 | ... | Bn) != 0)` for bit patterns `B1`, `B2`, ..., `Bn`.